internal/bminventory/inventory.go

```diff
@@ -343,6 +342,11 @@ func getImageName(clusterID, id strfmt.UUID) string {
 	return fmt.Sprintf("discovery-image-%s-%s", clusterID.String(), id)
 }
 
+func logAndGenerateError(id int32, errStr string) *models.Error {
```

Comment:
maybe it should be 
```suggestion
func logAndGenerateError(id int32, err error) *models.Error {
```
Then when you are using it you can use errors.Wrap
there are some places that you didn't log the original error. 

Another solution is maybe add the log to `generateError` but then you will need to pass context or logger.
@ronniel1 what do you think? 
if are already using generateErr then maybe the log should be there. and we will not need to log and create error separately.

Reply:
Already suggested how to handle ir. See https://github.com/filanov/bm-inventory/pull/186